Game.java

```diff
@@ -0,0 +1,76 @@
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.*;
+
+public class Game extends JFrame
+{
+    Board gameBoard;
+    int difficulty;
+    AutoSolver t;
+
+    public Game(int difficulty)
```

Comment:
I see it happens a lot so I'll write it one time.
1) A Java function should be a maximum of 30-40 lines, your function can be split to 3 or 4 small ones and the code will look a lot better
2) Try using const or maybe use enums(The best!)
3) There is a Java convention that you type '{' after class/function/if/else without the line break, It is not an ugly C# language.
4) I think you should try using generics like List<int, int>, I find it more convenient.
5) You have a lot of for loops, there is a beautiful thing since Java 8 called **Stream**, check it out
6) Try reading to code conventions for Java for prettier code
7) Use small classes and function unless they have a lot of logic(you don't have it here)

Beside that nice job! I love it that you keep learning and develop your skills.